### PR TITLE
#28: Add optional cache lifetime for checks

### PR DESCRIPTION
**Pull request by blackandred**
- Accidentialy made from issue_20 branch. Should be then merged after issue_20

See: #28 for feature description

### CODE REVIEW

File: infracheck/infracheck/exceptions.py [hunk 1]
```diff
@@ -5,6 +5,10 @@ class InfracheckException(Exception):
     pass
 
 
+class CheckNotReadyShouldBeSkippedSignal(InfracheckException):
+    pass
```
Comment (LazyScoundrel) [inline infracheck/infracheck/exceptions.py:9]: Is this an exception? (Do I understand this correctly?) If so, should it not be served in any way?

Comment (blackandred) [reply infracheck/infracheck/exceptions.py:9]: What do you mean by served?

Comment (blackandred) [reply infracheck/infracheck/exceptions.py:9]: It is raised there https://github.com/riotkit-org/infracheck/pull/29/files#diff-e58f93086ebe2f552c111b2ae8ebd2f0e974888c200d0bcec992678f32559ab5R61

And handled there: https://github.com/riotkit-org/infracheck/pull/29/files#diff-e58f93086ebe2f552c111b2ae8ebd2f0e974888c200d0bcec992678f32559ab5R121

Used to send a signal to parent context that performing the check should be skipped (reason: already in cache)

**Comment by blackandred** [comment]
Missing docs - forgot about them... :facepalm:

**Comment by blackandred** [comment]
Added docs.

**Review by LazyScoundrel — APPROVED** [review]

**Pull request merged**